components/profile/history/filter-tab.js

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useState } from 'react'
+import { Card, CardHeader, Divider, Checkbox, FormGroup, FormControlLabel } from '@mui/material'
+const FilterTab = (props) => {
+  const { option, callback } = props
+  const [actorChecked, setActorChecked] = useState(option.actors.map((actor) => true))
+  const [actionChecked, setActionChecked] = useState(option.actions.map((action) => true))
+  const [targetChecked, setTargetChecked] = useState(option.targets.map((target) => true))
+  const handleCheckActor = (event) => {

```

Comment:
- You are calculate your internal state by props. Consider using `getDerivedStateFromProps` => https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops.

- But let me remind you, it is not encouraged to calculate state using props, it is an anti-pattern, and it is rarely used. See docs by react: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html.

Reply:
this shit is useful bro, tfyh